### ISSUE TITLE
example/listen/doc.go is confusing

### ISSUE THREAD

**Issue by jrwren** [lib/pq#720]
Where is the NOTIFY statement or pg_notify() call?

**Comment by rutsky** [comment]
Apparently this is not implemented in pq. As a workaround you can execute `NOTIFY channel_name` statement as regular SQL query.

**Comment by silasdavis** [comment]
Yes this is very confusing. I think it at least needs a remark saying that `NOTIFY getwork` happens off stage. I was new to the mechanism and trying understand why you would ever a notification on `l.Notify`. The polling also seems like a bit of a distraction. I note there was a previous issue with the remark:

I don't oppose adding an example of how to use NOTIFY on that page, but I think it should be separate of the listener example.  Normally you wouldn't use LISTEN/NOTIFY to communicate within the same application (as you have channels at your disposal).

_Originally posted by @johto in https://github.com/lib/pq/issues/256#issuecomment-43178879_

I think it is better to show both sides of the interaction than to worry about whether the contrived example is too contrived.

**Comment by Daegalus** [comment]
I hit this same problem. I was trying to incorporate a simple PubSub with Notify/Listen and could not for the life of me find any information on how to send a NOTIFY. Now I know i have to do it manually. But a simple `pq.SendNotification(channel, payload)` would be more than enough. If I have some time I will do a pull request on this.

**Comment by manuganji** [comment]
> Package listen is a self-contained Go program which uses the LISTEN / NOTIFY
> mechanism to avoid polling the database while waiting for more work to arrive.
> 

I find it confusing that the document opens with "to avoid polling" but there's clearly polling with ` time.Sleep` and a `select` query. Why is this not polling?

https://github.com/lib/pq/blob/b2901c7946b69f1e7226214f9760e31620499595/example/listen/doc.go#L3
